lib/barzer/barzer_objects.py

```diff
@@ -0,0 +1,222 @@
+        else:
+            return (lo, hi)
+
+    def get_as_single_number(self, num_type=float):
+        pair = self.get_as_type_pair(the_type=num_type)
+        if pair[0] is None or pair[1] is None:
+            return None
+        else:
+            return (pair[0] + pair[1])/2
+
+class EVR(EntityBase):
+    """ entity value range """
+    type_name = "evr"
+    def __init__(self, data):
+        super(EVR, self).__init__(data)
+        self.ent = Entity(data['ent'])
+        self.values = [BeadFactory.make_bead_from_dict(x) for x in data.get('values', [])]
+
+    @property
+    def eclass(self):
+        return self.ent
+
+    @property
+    def subclass(self):
+        return self.subclass
+
+    @property
+    def id(self):
+        return self.id
+
+    def __str__(self):
+        return '{}:({} [{}])'.format(
+            self.type_name, str(self.ent), ','.join(str(x) for x in self.values))
+
+    def ent_id(self):
+        return self.ent.id
+
+    def iterate_type(self, the_type=Number):
+        """ yields all values of type the_type.
+        NOTE: the_type can also be a tuple
+        """
+        for v in self.values:
+            if isinstance(v, the_type):
+                yield v
+
+class ERC(EntityBase):
+    type_name = "erc"
+    """ entity range combo """
+    def __init__(self, data):
+        super(ERC, self).__init__(data)
+        self.ent = Entity(data['ent'])
+        self.range = Range(data.get('range'))
+
+    def __str__(self):
+        return '{}:({}{})'.format(self.type_name, str(self.ent), self.range)
+
+    def ent_id(self):
+        return self.ent.ent_id()
+
+class Time(ValueBead):
+    type_name = "time"
+    """ time of day """
+
+
+class Timestamp(Bead):
+    type_name = "timestamp"
+    """ """
+    def __init__(self, data):
+        super(Timestamp, self).__init__(data)
+        self.date, self.time = data['date'], data['time']
+
+    def __str__(self):
+        return '{}:{} {}'.format(self.type_name, self.date, self.time)
+
+class Date(ValueBead):
+    type_name = "date"
+    """ """
+
+class BeadFactory(object):
+    NAME_TYPE = {
+        'token': Token,
+        'fluff': Fluff,
+        'number': Number,
+        'punct': Token,
+        'range': Range,
+        'date': Date,
+        'timestamp': Timestamp,
+        'entity': Entity,
+        'erc': ERC,
+        'evr': EVR,
+    }
+
+    @classmethod
+    def make_beads_from_barz(cls, barz):
+        result = list()
+        for x in barz.get('beads'):
+            b = cls.make_bead_from_dict(x)
+            if b:
+                result.append(b)
+        return result
+
+    @classmethod
+    def make_bead_from_dict(cls, data, default_type=None):
+        the_type = data.get('type', default_type)
+        if the_type == 'entlist':
+            return [Entity(x) for x in data['data']]
+
+        bead_type = cls.NAME_TYPE.get(the_type)
+        if bead_type:
+            try:
+                return bead_type(data) if bead_type else None
```

Comment:
I suppose just return bead_type(data) here as we already checked that bead_type is not None